app/src/main/java/com/pajato/android/gamechat/common/adapter/ListAdapter.java

```diff
@@ -191,25 +195,48 @@ private View getView(final ViewGroup parent, final int resourceId) {
         return result;
     }
 
-    /** Update the optional icon in the given holder based on the specified item */
-    private void setOptIcon(final ItemListViewHolder holder, final ListItem item) {
+    /** Update the end icon in the given holder based on the specified item */
+    private void setEndIcon(final ItemListViewHolder holder, final ListItem item) {
         switch (item.type) {
             case chatGroup:
             case expGroup:
-                // Set an optional icon ONLY if the group is not the 'me' group (user cannot leave
+                // Set an end icon ONLY if the group is not the 'me' group (user cannot leave
```

Comment:
The code in these case clauses could benefit from using procedural abstractions, iMHO but only barely, fwiw.